### DISCUSSION TITLE
Sonar scanner plugin

### DISCUSSION THREAD

**Discussion by ppllaxxa** [sbt/sbt#7563] [Q&A]
Hi,

I am wondered that there is no supported sonar scanner plugin for SBT. I found only 2:
https://github.com/sonar-scala/sbt-sonar
https://github.com/sonar-scala/sonar-scala

It works fine with SonarQube <9, but doesn't work with SonarQube >9.
Maybe I missed something, or it is not usefull in most of the projects?

**Comment by eed3si9n** [comment]
I've only heard of SonarQube, but never seen any Scala project actually use it in real life. I feel like `-Xlint` and `-Xfatal-warnings` gets a bunch in Scala, and if you need stronger enforcement, it becomes somewhat codebase specific, so people start writing custom Scalafix rules or WartRemover warts.

**Comment by gitricko** [comment]
You might wanna try sonarless which automates the above in just a few steps.

Local install:

Install CLI: `curl -s "https://raw.githubusercontent.com/gitricko/sonarless/main/install.sh" | bash`

Scan current directory: `sonarless scan`

Get results: `sonarless results`

Check out the automation

https://github.com/gitricko/sonarless

It even have a GitHub Action for GITHUB CI. The nice thing about this is that you don't need a host sonarqube
